### PR TITLE
handle empty fused norm result

### PR DESCRIPTION
**Pull request by qiuxin2012**
## Description

handle empty fused norm result when fused norm failed.

### 1. Why the change?

handle empty fused norm result when fused norm failed.

### 2. User API changes

None

### 4. How to test?
- [ ] Manual tes,t, force kernel return empty tensor

### CODE REVIEW

**Review by rnwang04 — APPROVED** [review]

File: python/llm/src/bigdl/llm/transformers/models/baichuan2.py [hunk 1]
```diff
@@ -50,24 +50,25 @@ def baichuan_13b_rms_norm_forward(self, hidden_states):
...
                                                        [self.weight.size(0)],
                                                        self.weight,
                                                        None,
                                                        self.epsilon)
+            # if nelement == 0, means fused norm failed, go back to python implement.
+            if result.nelement != 0:
+                return result
         else:
-            hidden_states = torch.ops.torch_ipex.fast_rms_norm(hidden_states,
+            return torch.ops.torch_ipex.fast_rms_norm(hidden_states,
```
Comment (jason-dai) [inline python/llm/src/bigdl/llm/transformers/models/baichuan2.py:62]: We need to call `fused_rms_norm` in both cases to support `xpu:1`, yes? @lalalapotter

File: python/llm/src/bigdl/llm/transformers/models/baichuan2.py [hunk 2]
```diff
@@ -50,24 +50,25 @@ def baichuan_13b_rms_norm_forward(self, hidden_states):
...
         if get_ipex_version() <= "2.0.110+xpu":
             import linear_q4_0
-            hidden_states = linear_q4_0.fused_rms_norm(hidden_states,
+            result = linear_q4_0.fused_rms_norm(hidden_states,
                                                        [self.weight.size(0)],
                                                        self.weight,
                                                        None,
                                                        self.epsilon)
+            # if nelement == 0, means fused norm failed, go back to python implement.
+            if result.nelement != 0:
```
Comment (jason-dai) [inline python/llm/src/bigdl/llm/transformers/models/baichuan2.py:59]: do you need to check for `2.1`?

File: python/llm/src/bigdl/llm/transformers/models/baichuan2.py (hunk 1 above)
Comment (lalalapotter) [reply python/llm/src/bigdl/llm/transformers/models/baichuan2.py:62]: Yes, for ipex 2.1, `torch.ops.torch_ipex.fast_rms_norm` is not work normally for `xpu:1`.

File: python/llm/src/bigdl/llm/transformers/models/baichuan2.py (hunk 2 above)
Comment (qiuxin2012) [reply python/llm/src/bigdl/llm/transformers/models/baichuan2.py:59]: No, if can not be fused, fast_rms_norm will go to another native implement. We don't have the native implement, so we go back to python.

Comment (jason-dai) [reply python/llm/src/bigdl/llm/transformers/models/baichuan2.py:59]: You should use fused_rms_norm for both 2.0 & 2.1

**Pull request merged**